### PR TITLE
Dynamic host attributes

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: packages/compiler/src/render3/view/compiler.ts [hunk 1]
```diff
@@ -217,7 +205,8 @@ export function compileComponentFromMetadata(
   const firstSelector = selector && selector[0];
 
   // e.g. `attr: ["class", ".my.app"]`
-  // This is optional an only included if the first selector of a component specifies attributes.
+  // This is optional an only included if the first selector of a component specifies
+  // attributes.
```
Comment (AndrewKushnir) [inline packages/compiler/src/render3/view/compiler.ts]: Formatting issue, could you plz revert?

File: packages/compiler/src/render3/view/compiler.ts [hunk 2]
```diff
@@ -325,7 +314,8 @@ export function compileComponentFromMetadata(
     definitionMap.set('changeDetection', o.literal(changeDetection));
   }
 
-  // On the type side, remove newlines from the selector as it will need to fit into a TypeScript
+  // On the type side, remove newlines from the selector as it will need to fit into a
+  // TypeScript
```
Comment (AndrewKushnir) [inline packages/compiler/src/render3/view/compiler.ts]: Formatting issue, could you plz revert? (see a few other places below as well)

File: packages/compiler/src/render3/view/compiler.ts [hunk 3]
```diff
@@ -412,34 +406,7 @@ export function compileComponentFromRender2(
...
-      listeners: summary.hostListeners,
-      properties: summary.hostProperties,
-    },
-    inputs: directive.inputs,
-    outputs: directive.outputs,
-    usesInheritance: false,
-    exportAs: null,
-    providers: directive.providers.length > 0 ? new o.WrappedNodeExpr(directive.providers) : null
-  };
+  throw new Error('unsupported');
```
Comment (AndrewKushnir) [inline packages/compiler/src/render3/view/compiler.ts]: Could you please confirm if this is an intended change? May be add a comment to describe it?

File: packages/compiler/src/render3/view/compiler.ts [hunk 4]
```diff
@@ -855,36 +832,66 @@ const enum HostBindingGroup {
...
+  for (const key of Object.keys(host)) {
     const value = host[key];
     const matches = key.match(HOST_REG_EXP);
+
     if (matches === null) {
-      attributes[key] = value;
+      switch (key) {
+        case 'class':
+          if (typeof value !== 'string') {
+            throw new Error(`Class binding must be string`);
```
Comment (AndrewKushnir) [inline packages/compiler/src/render3/view/compiler.ts]: May be add a couple tests to check this condition (and other conditions where we throw below)?

**Review by AndrewKushnir — APPROVED** [review]
LGTM, just left a few comments.

File: packages/compiler/src/render3/view/compiler.ts (hunk 4 above)
Comment (alxhub) [reply packages/compiler/src/render3/view/compiler.ts]: I added TODOs to add `ts.Diagnostic`s. That's going to be a tricky refactor as the `parseHostBindings` operation doesn't return errors at all currently. We should add tests whenever we have proper error support.

Comment (AndrewKushnir) [reply packages/compiler/src/render3/view/compiler.ts]: Sounds good, thanks!

**Comment by AndrewKushnir** [comment]
Thanks for addressing comments @alxhub! 👍

**Comment by alxhub** [comment]
Caretaker: I am the approver for fw-compiler

**Pull request closed without merging**